### PR TITLE
removed weekly schedule header impelementation for #1869

### LINKED ISSUE #1869: drop table size check for WeeklySchedule
As discussed on today's call:
```
#234asdfasdfasdfasdfasdfljksdfklajslkdfjaskdfjkasdk;lfajksldfjasdkfjas;dfkjasklf;asdkfaksdjf;asdjkfajskd
double  tab1(3,5) #test:
#
mon:0:0:10              -       3       1       -   
tue,thu:20:30:59        123     -       45      -   
wed 
```
the second line can be dropped/deprecated and the number of rows check can be dropped too. There should still be a check that verifies whether all lines have the same number of columns.

### PR DESCRIPTION
**Pull request by Mathadon**
closes #1869

### CODE REVIEW

File: IBPSA/Resources/C-Sources/WeeklySchedule.c [hunk 1]
```diff
@@ -378,10 +312,18 @@ void* weeklyScheduleInit(const int tableOnFile, const char* name, const double t
...
-            weeklyScheduleFreeInit(scheduleID);
-            weeklyScheduleFree(scheduleID);
-            ModelicaFormatError("Too many columns on row %i when reading weekly schedule '%s'.", line, name);
+          if (tokensInFirstLine != 0 && tokensInLine + 1 > tokensInFirstLine){
+              weeklyScheduleFreeInit(scheduleID);
+              weeklyScheduleFree(scheduleID);
+              ModelicaFormatError("Too many columns on data line %i when reading weekly schedule '%s'. %i instead of %i.", line, name, tokensInLine + 1, tokensInFirstLine);
+          }else if (tokensInLine >= n_reservedColumns) { /* This code should only be reached upon passing the first line of data */
+            n_reservedColumns += allocSize;
+            scheduleID->rules[rule_i]->data = (double*)realloc(sizeof(double), n_reservedColumns);
```
Comment (mwetter) [inline IBPSA/Resources/C-Sources/WeeklySchedule.c:321]: This is wrong, check the function signature of `realloc`.

File: IBPSA/Resources/C-Sources/WeeklySchedule.c [hunk 2]
```diff
@@ -501,10 +437,14 @@ void weeklyScheduleFreeInit(void * ID) {
 void weeklyScheduleFree(void * ID) {
   WeeklySchedule* scheduleID = (WeeklySchedule*)ID;
 
+  if (ID == NULL) /* Otherwise OM segfaults when IBPSA.Utilities.IO.Files.Examples.WeeklySchedule triggers an error */
+    return;
+
   int i;
```
Comment (mwetter) [inline IBPSA/Resources/C-Sources/WeeklySchedule.c:443]: `int i` must be declared on top of the function according to C90.

**Review by mwetter — CHANGES_REQUESTED** [review]

File: IBPSA/Resources/C-Sources/WeeklySchedule.c (hunk 2 above)
Comment (Mathadon) [reply IBPSA/Resources/C-Sources/WeeklySchedule.c]: fixed

File: IBPSA/Resources/C-Sources/WeeklySchedule.c (hunk 1 above)
Comment (Mathadon) [reply IBPSA/Resources/C-Sources/WeeklySchedule.c]: Good catch.. thanks! I did not realise this code was untested.  I tested it using a long table and spotted another implementation error that is now fixed!

**Review by mwetter — APPROVED** [review]

**Pull request merged**